core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -912,6 +912,7 @@ public long getLastWriteNS() {
     private void markLastWrite(Engine.Operation op) {
         lastWriteNS = op.startTime();
         if (active.getAndSet(true) == false) {
+            engineConfig.setFlushWhenLastMergeFinished(false);
```

Comment:
It's weird what we are communicating this boolean value via `EngineConfig`?  It's not really a "config" thing but rather a live state of the engine?  Maybe we should move `active` boolean down to `InternalEngine` and you just use that?  This can be a later change though...